test/Form.test.js

```diff
@@ -51,25 +53,26 @@ describe('Form.test.js', () => {
     })
 
     it('is not called if value is empty (trimmed)', next => {
-      cmp.vm.inputValue = '   '
+      cmp.setData({ inputValue: '   ' })
       cmp.vm.$nextTick(() => {
```

Comment:
From blog post:

> We change the inputValue, so the console.log spy should be called, right? Well, if you run it, you'll notice that is not! WTF??? Wait, there is an explanation for this: unlike computed properties, watchers are deferred to the next update cycle that Vue uses to look for changes. So, basically, what's happening here is that console.log is indeed called, but after the test has finished.

> To solve this, we need to use the **vm.$nextTick** function to defer code to the next update cycle. But if we write:


But if I remove the use of `$nextTick` in all tests, then they will pass successfully. So can this `$nextTick` be removed altogether?

Where did you get the information that?

> **unlike computed properties, watchers are deferred to the next update.**

Maybe earlier it was? But now there is apparently no need for `$nextTick`, please explain this. In which case, it will require a change in the text of the book/article.

сс @alexjoverm

Reply:
Can you try to write an assertion that should fail within the nextTick? Something like:

```
vm.$nextTick(() => expect(false).toBeTruthy())
```

If it passes, it's because what's inside is not called. That's maybe what's happening in your code.

Reply:
Oh wait, I see you're using setData. In that case is not needed, since that method already executes the nextTick inside :)

Reply:
It would be cool, if instead of removing that text on the article, let's better keep it as it is an later show how the setData (and the other of vue-test-utils) do that for you.

In that way, the user know how it's done raw and how vue-test-utils makes it easier for you